src/main/java/trackmodel/TrackModel.java

```diff
@@ -1548,7 +1548,11 @@ public int getTrainPassengers(int trainId) {
             // load waiting passengers
             Double passengers = this.waitingPassengers.get(this.getTrainBlock(trainId));
             if (passengers != null) {
-                int loaded = this.random.nextInt(Math.min((int) passengers.doubleValue(), train.getMaxPassengers()));
+                int bound = Math.min((int) passengers.doubleValue(), train.getMaxPassengers());
+                int loaded = 0;
+                if (bound > 0) {
+                    loaded = this.random.nextInt();

```

Comment:
Need to add `bound` as a parameter to `nextInt()`. With no params, [`nextInt()`](https://docs.oracle.com/javase/8/docs/api/java/util/Random.html#nextInt--) returns a uniformly distributed value from 0 to 2<sup>32</sup>.